Updated sidebar

Achu-Worifung opened the pull request:
updated the sidebar to match the Figma design.

maishaSupritee reviewed (changes requested):
- Need more space between the icon and the title in the header.
![image](https://github.com/user-attachments/assets/a725cb99-08a4-46d9-9958-abf5bf7ddaed)

- User icon needs to be filled with white, settings and notifications icon you have outlined but we want filled versions
Your one:
![image](https://github.com/user-attachments/assets/d01bf776-e018-4184-895b-5a0aa648e3ef)

Figma:
![image](https://github.com/user-attachments/assets/ecf32109-09e1-41e6-80aa-fa6bb17ed997)

- The notifications modal opens up if I hover on the notifications icon. It should open when I click on the icon only.
- Why do you have the toggle component from shadcn? You are not using it anywhere.
- The sign in, settings and notifications should all just be buttons. Why are they wrapped in HTML hyperlinks and spans?
![image](https://github.com/user-attachments/assets/62af65d1-f4eb-4f95-b94f-27aadf4e5b3e)

- Why did you change the theme-toggle file? This was not part of your task, and what changes did you make and why?

Achu-Worifung commented:
<img width="763" alt="Screenshot 2025-03-08 144408" src="https://github.com/user-attachments/assets/46e0d1e4-3a31-44f5-bdd6-09aab9a1880b" />

added more spacing between page header and icon.
filled in the icons in the header
changed notification from hover card to popover card. ( now drops down on click)
changed sign in and settings to button not the notification as doing so triggered a hydration error due to the trigger being a button.
refactored header and got rid of unnecessary nesting. 

I don't believe that I changed the theme toggle file created by Dat. I made a new one that toggled the theme without using the dropdown. Once I reached out to you and you said it wasn't my task i abandoned it and simply failed to delete it.

Important.
I could not achieve the fill for the settings icon using the fill and stroke, so i copied the jsx from lucide and made changes to it. Please let me know if this is acceptable or not.
